### DISCUSSION TITLE
problem with marble build

### DISCUSSION THREAD

**Discussion by Saleh-At** [albar965/littlenavmap#935] [Q&A]
I used step by step your [instruction](https://github.com/albar965/littlenavmap/wiki/Compiling#compile-marble) for windows 11 in `lnm/1.1` branch.
`cmake ...` is ok complied and for `mingw32-make -j2` or `mingw32-make -j4` showed errors for `#include` of `QApplication` or `Qobject` or `c++11`.
I installed the latest version of `cmake` and Qt 5.14.1.
How to fix it?
Best regards

* [log ](https://docs.google.com/document/d/1vGwCkfPowoGz-6kJJodsQu5ypzMeplBNQSS-lv7M-w8/edit?usp=sharing)file

**Comment by albar965** [comment]
You cmake apparently decided to use MSVC++ to compile the project. I'm not sure if this works for Marble but atools and LNM cannot be compiled with VC. You use the MinGW toolchain to compile but the compiler is MSVC.

I compile Marble on Windows using this:
```
cd build-marble-release

set PATH=%PATH%;C:\Qt\5.15.2\mingw81_32\bin;C:\Qt\Tools\mingw810_32\bin

cmake -Wno-dev -G "MinGW Makefiles" -DQTONLY=TRUE -DQT5BUILD=TRUE -DWITH_quazip=FALSE -DWITH_libshp=FALSE -DWITH_libgps=FALSE -DWITH_libwlocate=FALSE -DWITH_QtLocation=FALSE -DWITH_liblocation=FALSE -DCMAKE_INSTALL_PREFIX=%APROJECTS%\Marble-release -DCMAKE_BUILD_TYPE=Release ..\marble

mingw32-make -j4

mingw32-make install
```
What I have installed on the Windows build system:
![grafik](https://user-images.githubusercontent.com/15254956/189378793-9a250b3a-3fd0-43ce-9260-4af4e54d628e.png)

Alex

**Comment by Saleh-At** [comment]
Thx, I did as you send but I have same problems : [log2](https://docs.google.com/document/d/1vZk0IO5xPFld1uwHv9-ciidkdGqPIswJteUttgNuCg8/edit?usp=sharing)
I thought the compiler can't find lib of qt such as `Qt::Core`.

**Comment by albar965** [comment]
Do you use a too old compiler? No idea why this pops up:
`#  error "Qt requires C++11 support"`

The problem is that Marble is not my project (I happy to compile this myself) and I have not much clue about cmake. I use qmake for all my projects.

**Comment by Saleh-At** [comment]
I was confused, I used that version of pointed in your screenshot. And clone your Marble repo.
How to compile [this](https://github.com/albar965/marble/tree/lnm/1.1) by qmake? cause I am new to cmake.
And can share your build?
thx.

**Comment by albar965** [comment]
Marble uses cmake and all LNM projects use qmake. This is not a such big problem since Marble is rarely changed and even LNM 2.6 to 2.8 can run with the same current Marble version.

Sharing the build is no problem.

I can upload my current Marble Windows build if it helps. But this is a 32 bit build since I need this to be compatible with old SimConnect versions.

**Comment by Saleh-At** [comment]
ok thanks, so send me your builds in drobox or anywhere. (I don't use SimConnect !)
- I want to "Little Nav App" in 64-bit ver, compiling marble or other repos with 32-bit compiler is ok?

**Comment by albar965** [comment]
> * I want to "Little Nav App" in 64-bit ver, compiling marble or other repos with 32-bit compiler is ok?

You can compile all of LNM and Marble in 64 bit (Linux and macOS are 64 bit) but I don't have such a build. For now I stick with a 32 bit build to be compatible with old SimConnect versions. I might do a 64 bit Windows build in the future but this will be usable only with MSFS and X-Plane (causing a lot of confusion for users).

**Comment by Saleh-At** [comment]
Can with this build, use in `qmake` project (`.pro`). Is there any tutorial to add to it? 
I read the main [marble site](https://marble.kde.org/) for `.qml` file, But I confused.
I want a tutorial like [qnanopainter](https://github.com/QUItCoding/qnanopainter#usage).
thx alot

**Comment by albar965** [comment]
> Can with this build, use in `qmake` project (`.pro`). Is there any tutorial to add to it?

Not sure if I understand. Marble needs cmake and LNM qmake. I cannot do anything about this.

> I read the main [marble site](https://marble.kde.org/) for `.qml` file, But I confused.

Ignore the Marble main site. Not sure what you mean by the `.qml` file.

> I want a tutorial like [qnanopainter](https://github.com/QUItCoding/qnanopainter#usage).

This is the [BUILD.txt](https://github.com/albar965/littlenavmap/blob/master/BUILD.txt).

**Comment by albar965** [comment]
Converted to a discussion.
